clippy_lints/src/strings.rs

```diff
@@ -467,6 +468,49 @@ impl<'tcx> LateLintPass<'tcx> for StringToString {
     }
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for calling `str::split` with the ASCII space character instead of `str::split_whitespace`.
+    ///
+    /// ### Why is this bad?
+    /// `split` using the ASCII space character does not handle cases where there is a different space character used. `split_whitespace` handles all whitespace characters.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// "A B C".split(' ');
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// "A B C".split_whitespace();
+    /// ```
+    #[clippy::version = "1.81.0"]
+    pub SPLIT_WITH_SPACE,
+    pedantic,
+    "using `str::split` with the ASCII space character instead of `str::split_whitespace`"
+}
+declare_lint_pass!(SplitWithSpace => [SPLIT_WITH_SPACE]);
+
+impl<'tcx> LateLintPass<'tcx> for SplitWithSpace {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &Expr<'_>) {
+        if let ExprKind::MethodCall(path, _split_recv, [arg], split_ws_span) = expr.kind
+            && path.ident.name == sym!(split)
+            && let ExprKind::Lit(lit) = arg.kind
+            && (matches!(lit.node, LitKind::Char(' '))
+                || matches!(lit.node, LitKind::Str(str, _) if str.as_str() == " "))
```

Comment:
There should be a check here that this isn't from a macro, (or at least, not from an external macro since the user can't change that, can use [`in_external_macro`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/lint/fn.in_external_macro.html) for this)